sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2690,6 +2690,18 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val GROUPING_WITH_UNION = buildConf("spark.sql.optimizer.grouping")
+    .doc("When true,enable grouping with union.")
```

Comment:
For text style. Please add a space after punctuation marks like , or .

For example, "When true,enable grouping with union." -> "When true, enable grouping with union."

There are several places with this style issue.